### DISCUSSION TITLE
[Wasm] undefined symbol: _ZNKSt3__220__vector_base_commonILb1EE20__throw_length_errorEv

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#10806] [Q&A] [answered]
[_From Discord:_](https://discord.com/channels/372137812037730304/550416151172087808/1055421587777671198)

Hello everyone. Discovered a strange error while developing our new project on Uno. It's using the latest stable versions of all nuget packages and building WinUI head just fine. But building WASM head fails with this:

```
1>    [00:04:56.2731375] error: undefined symbol: _ZNKSt3__220__vector_base_commonILb1EE20__throw_length_errorEv (referenced by top-level compiled C/C++ code)
1>    [00:04:56.2731485] warning: Link with '-sLLD_REPORT_UNDEFINED' to get more information on undefined symbols
1>    [00:04:56.2731521] warning: To disable errors for undefined symbols use '-sERROR_ON_UNDEFINED_SYMBOLS=0'
1>    [00:04:56.2731544] warning: __ZNKSt3__220__vector_base_commonILb1EE20__throw_length_errorEv may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
1>    [00:04:56.2731569] Error: Aborting compilation due to previous errors
1>    [00:04:56.2731592] emcc: error: 'C:/Users/X/.uno/emsdk/emsdk-3.1.12/emsdk/node/14.18.2_64bit/bin/node.exe C:\Users\X\.uno\emsdk\emsdk-3.1.12\emsdk\upstream\emscripten\src\compiler.js C:\Users\X\AppData\Local\Temp\tmphuz1951u.json' failed (returned 1)
1>    [00:04:56.2732811] ninja: build stopped: subcommand failed.
1>    E:\NugetCache\uno.wasm.bootstrap\7.0.11\build\Uno.Wasm.Bootstrap.targets(207,3): error : System.Exception: Failed to run emscripten.
```

I tried running uno check and emcc check, both look good. Clearing emscripten cache changed nothing too.
The strange thing is our older Uno projects (like 1-2 years old, we're long time Uno admirers :unoheart:) still build WASM head fine 🤔

**Comment by jeromelaban** [comment, marked as answer]
Hi! Thanks for being a long time Uno user!

This generally means that you're using a native dependency that has not been updated to run on the latest emscripten. One example is SkiaSharp (it must be 2.88.3 with the bootstrapper 7.0.x).

**Reply by gevart** [reply]
That's exactly what the problem was!
I was using LiveCharts for Uno and their nuget depends on SkiaSharp 2.88.2. Updated transient package to 2.88.3 and now it compiles. Thanks a lot!

**Reply by rabejens** [reply]
I had the same problem, for me, I had to explicitly install `SkiaSharp.Views.Uno.WinUI` in my WASM head.
